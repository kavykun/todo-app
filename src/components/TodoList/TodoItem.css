.todo-item {
  display: flex;
  width: 100%;
  background-color: #f3f3f3;
  margin-top: 10px;
  height: 26px;
}

@media only screen and (max-width: 600px) {
  .todo-item {
    margin-top: 20px;
  }
}

.todo-item-description {
  font-size: 12px;
  line-height: 24px;
  margin-left: 4px;
}

.todo-item > .checkbox {
  margin-top: 5px !important;
}

.todo-item > .date {
  margin: 3px 3px 3px auto;
  padding: 3px;
  font-size: 10px;
}

.overDue {
  background-color: #ffcccc;
}

.completed {
  background-color: #cbffcc;
}

.completed > .todo-item-description {
  text-decoration: line-through;
}
