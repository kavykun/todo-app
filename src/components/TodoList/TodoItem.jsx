import React from 'react';
import PropTypes from 'prop-types';

import Checkbox from '../../common/Checkbox';
import Date from '../../common/Date';

import './TodoItem.css';

const TodoItem = (props) => {
  const {
    completed,
    overDue,
    todo: { id, description, dueDate },
    handleInputChange,
  } = props;

  return (
    <div
      className={`todo-item ${completed ? 'completed' : ''} ${
        overDue ? 'overDue' : ''
      }`}
      key={id}
    >
      <Checkbox
        name="todo-item-checkbox"
        handleInputChange={() => handleInputChange(id)}
        checked={completed}
      />
      <div className="todo-item-description">{description}</div>
      <Date date={dueDate} />
    </div>
  );
};

TodoItem.defaultProps = {
  completed: false,
  overDue: false,
  handleInputChange: () => {},
  todo: {
    id: '',
    description: '',
    dueDate: ' ',
  },
};

TodoItem.propTypes = {
  completed: PropTypes.bool,
  overDue: PropTypes.bool,
  handleInputChange: PropTypes.func,
  todo: PropTypes.shape({
    id: PropTypes.string,
    description: PropTypes.string,
    dueDate: PropTypes.string,
  }),
};

export default TodoItem;
