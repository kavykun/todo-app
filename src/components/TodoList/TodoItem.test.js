import { render, fireEvent, screen } from '@testing-library/react';

import TodoItem from './TodoItem';

test('Should render a single todo item', async () => {
  const mockProps = {
    completed: true,
    overDue: false,
    handleInputChange: jest.fn(),
    todo: {
      id: '',
      description: '',
      dueDate: '05/05/2005',
    },
  };
  render(<TodoItem {...mockProps} />);
  expect(screen.getByRole('checkbox'));
});

test('Should handle click event if checkbox is pressed in todo item', async () => {
  const mockProps = {
    completed: false,
    overDue: true,
    handleInputChange: jest.fn(),
    todo: {
      id: '',
      description: '',
      dueDate: '05/05/2005',
    },
  };
  render(<TodoItem {...mockProps} />);
  fireEvent.click(screen.getByRole('checkbox'));
  expect(mockProps.handleInputChange).toHaveBeenCalledTimes(1);
});
