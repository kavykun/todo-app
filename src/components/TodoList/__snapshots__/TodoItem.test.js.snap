// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="spinner"
  >
    <div
      class="bounce1"
    />
    <div
      class="bounce2"
    />
    <div
      class="bounce3"
    />
  </div>
</DocumentFragment>
`;
